packages/kilt-did/src/types.ts

```diff
@@ -0,0 +1,104 @@
+import type { IIdentity, SubmittableExtrinsic } from '@kiltprotocol/types'
+import { u64 } from '@polkadot/types'
+import { AnyNumber } from '@polkadot/types/types'
+import { KeyId } from './types.chain'
+
+export type KeypairType = string
+
+export interface IPublicKey {
+  publicKey: Uint8Array
+  type: KeypairType
+  // do we need the identifier?
+}
+
+export type IKeyPair = IPublicKey
+
+export type ISigningKey = IPublicKey
+export type IEncryptionKey = IPublicKey
+
+export interface ISigningKeyPair extends ISigningKey {
+  sign: (message: string | Uint8Array) => Uint8Array
+}
+
+export interface IEncryptionKeyPair extends IEncryptionKey {
+  decrypt: (
+    message: Uint8Array,
+    additionalData: Record<string, unknown>
+  ) => Uint8Array | null
+}
+
+// TODO: should this use the same keys as the chain type?
+export interface KeySet {
```

Comment:
Since this does not include the past attestation keys (now simple verification keys), perhaps it would be better to call it `ActiveKeySet`.

Reply:
just realised I'm not actually using this any more. So I may just remove this definition.

Reply:
Ok, if not used we can remove it. Otherwise I think it would be better to rename it.